android/src/main/java/com/rnziparchive/RNZipArchiveModule.java

```diff
@@ -303,82 +304,70 @@ public void onCopyProgress(long bytesRead) {
+      ZipParameters parameters = new ZipParameters();
+      parameters.setCompressionMethod(Zip4jConstants.COMP_DEFLATE);
+      parameters.setCompressionLevel(Zip4jConstants.DEFLATE_LEVEL_NORMAL);
+
+      ProcessZip(fileOrDirectory, destDirectory, parameters, promise);
+
+    } catch (Exception ex) {
+      promise.reject(null, ex.getMessage());
+      return;
+    }
+  }
+
+  @ReactMethod
+  public void zipWithPassword(String fileOrDirectory, String destDirectory, String password,
+      String encryptionMethod, Promise promise) {
+    try{
+
+      ZipParameters parameters = new ZipParameters();
+      parameters.setCompressionMethod(Zip4jConstants.COMP_DEFLATE);
+      parameters.setCompressionLevel(Zip4jConstants.DEFLATE_LEVEL_NORMAL);
+
+      String encParts[] = _encryptionMethod.split("-");
+
+      if (password != null && !password.isEmpty()) {
+        parameters.setEncryptFiles(true);
+        if (encParts[0].equals("AES")) {
+          parameters.setEncryptionMethod(Zip4jConstants.ENC_METHOD_AES);
+          if (encParts[1].equals("128")) {
+            parameters.setAesKeyStrength(Zip4jConstants.AES_STRENGTH_128);
+          } else if (encParts[1].equals("256")) {
+            parameters.setAesKeyStrength(Zip4jConstants.AES_STRENGTH_256);
+          } else {
+            parameters.setAesKeyStrength(Zip4jConstants.ENC_METHOD_STANDARD);
           }
+        } else if (_encryptionMethod.equals("STANDARD")) {
+          parameters.setEncryptionMethod(Zip4jConstants.ENC_METHOD_STANDARD);
+          Log.d(TAG, "Standard Encryption");
         } else {
-          fromDirectory = fileOrDirectory.substring(0, fileOrDirectory.lastIndexOf("/"));
-          filePaths.add(fileOrDirectory);
+          parameters.setEncryptionMethod(Zip4jConstants.ENC_METHOD_STANDARD);
+          Log.d(TAG, "Encryption type not supported default to Standard Encryption");
         }
+        parameters.setPassword(password);
       } else {
-        throw new FileNotFoundException(fileOrDirectory);
+        promise.reject(null, "Password is empty");
       }
-    } catch (FileNotFoundException | NullPointerException e) {
-      promise.reject(null, "Couldn't open file/directory " + fileOrDirectory + ".");
-      return;
-    }
 
-    try {
-      String[] filePathArray = filePaths.toArray(new String[filePaths.size()]);
-      new ZipTask(filePathArray, destDirectory, fromDirectory, promise, this).zip();
+      ProcessZip(fileOrDirectory, destDirectory, parameters, promise);
+
     } catch (Exception ex) {
       promise.reject(null, ex.getMessage());
       return;
     }
+
   }
 
-  @ReactMethod
-  public void zipWithPassword(final String fileOrDirectory, final String destDirectory, final String password,
-      final String encryptionMethod, final Promise promise) {
+  private void ProcessZip(final String fileOrDirectory, final String destDirectory, final ZipParameters parameters, final Promise promise) {
```

Comment:
It would be better to write method name in *lowerCamelCase* format, which is recommended by https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names

```suggestion
  private void processZip(final String fileOrDirectory, final String destDirectory, final ZipParameters parameters, final Promise promise) {
```